From ced2069da079be93b093f4274b5ef428e98870b5 Mon Sep 17 00:00:00 2001
From: kergoth <kergoth@a933c7a4-9bf7-0310-9141-a2e45189604d>
Date: Tue, 11 Nov 2008 21:27:07 +0000
Subject: [PATCH 16/30] BTN_TOUCH is only *required* when the device isn't able to measure pressure,
 since we can also infer a press/release from the pressure value.

Signed-off-by: Chris Larson <clarson@kergoth.com>


git-svn-id: svn://svn.berlios.de/tslib/trunk/tslib@64 a933c7a4-9bf7-0310-9141-a2e45189604d
---
 plugins/input-raw.c |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/plugins/input-raw.c b/plugins/input-raw.c
index 9e2ea60..d342dc7 100644
--- a/plugins/input-raw.c
+++ b/plugins/input-raw.c
@@ -94,13 +94,14 @@ static int check_fd(struct tslib_input *i)
 	we set it to constant 255. It's still controlled by BTN_TOUCH - when not
 	touched, the pressure is forced to 0. */
 
-	if (!(absbit[BIT_WORD(ABS_PRESSURE)] & BIT_MASK(ABS_PRESSURE)))
+	if (!(absbit[BIT_WORD(ABS_PRESSURE)] & BIT_MASK(ABS_PRESSURE))) {
 		i->current_p = 255;
 
-	if ((ioctl(ts->fd, EVIOCGBIT(EV_KEY, KEY_CNT), keybit) < 0) ||
-		!(absbit[BIT_WORD(BTN_TOUCH)] & BIT_MASK(BTN_TOUCH)) ) {
-		fprintf(stderr, "selected device uses is not a touchscreen (must support BTN_TOUCH events)\n");
-		return -1;
+		if ((ioctl(ts->fd, EVIOCGBIT(EV_KEY, KEY_CNT), keybit) < 0) ||
+			!(absbit[BIT_WORD(BTN_TOUCH)] & BIT_MASK(BTN_TOUCH)) ) {
+			fprintf(stderr, "selected device uses is not a touchscreen (must support BTN_TOUCH events)\n");
+			return -1;
+		}
 	}
 
 	if (evbit[BIT_WORD(EV_SYN)] & BIT_MASK(EV_SYN))
-- 
1.6.5.2

